src/VisualStudio/Core/Def/StackTraceExplorer/StackTraceExplorerToolWindow.cs

```diff
@@ -0,0 +1,112 @@
+    {
+        private bool _initialized;
+        public StackTraceExplorerRoot? Root { get; private set; }
+
+        public StackTraceExplorerToolWindow() : base(null)
+        {
+            Caption = ServicesVSResources.Stack_Trace_Explorer;
+            Content = new DockPanel
+            {
+                LastChildFill = true
+            };
+        }
+
+        public void InitializeIfNeeded(RoslynPackage roslynPackage)
+        {
+            if (_initialized)
+            {
+                return;
+            }
+
+            var workspace = roslynPackage.ComponentModel.GetService<VisualStudioWorkspace>();
+            var formatMapService = roslynPackage.ComponentModel.GetService<IClassificationFormatMapService>();
+            var formatMap = formatMapService.GetClassificationFormatMap(StandardContentTypeNames.Text);
+            var typeMap = roslynPackage.ComponentModel.GetService<ClassificationTypeMap>();
+            var threadingContext = roslynPackage.ComponentModel.GetService<IThreadingContext>();
+            var streamingFindUsagesPresenter = roslynPackage.ComponentModel.GetService<IStreamingFindUsagesPresenter>();
+
+            Root = new StackTraceExplorerRoot(new StackTraceExplorerRootViewModel(threadingContext, workspace, formatMap, typeMap, streamingFindUsagesPresenter))
+            {
+                HorizontalAlignment = HorizontalAlignment.Stretch,
+                VerticalAlignment = VerticalAlignment.Stretch
+            };
+
+            var contentRoot = (DockPanel)Content;
+            contentRoot.Children.Add(Root);
+
+            var contextMenu = new ThemedContextMenu();
+            contextMenu.Items.Add(new MenuItem()
+            {
+                Header = ServicesVSResources.Paste,
+                Command = new DelegateCommand(_ => Root.OnPaste()),
+                Icon = new CrispImage()
+                {
+                    Moniker = KnownMonikers.Paste
+                }
+            });
+
+            contextMenu.Items.Add(new MenuItem()
+            {
+                Header = ServicesVSResources.Clear,
+                Command = new DelegateCommand(_ => Root.OnClear()),
+                Icon = new CrispImage()
+                {
+                    Moniker = KnownMonikers.ClearCollection
+                }
+            });
+
+            contentRoot.ContextMenu = contextMenu;
+
+            _initialized = true;
+        }
+
+        public override void OnToolWindowCreated()
+        {
+            // Hide the frame by default when VS starts
+            if (Frame is IVsWindowFrame windowFrame)
+            {
+                windowFrame.CloseFrame((uint)__FRAMECLOSE.FRAMECLOSE_NoSave);
+            }
+        }
+
+        int IOleCommandTarget.Exec(ref Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
+        {
+            //if ((nCmdID & (uint)OLECMDID.OLECMDID_PASTE) != 0 ||
```

Comment:
I still need to follow up on what the correct way to handle paste here is. This will let the tool window forward all paste commands correctly since WPF doesn't always capture paste (it's based on control geometry and other factors)